catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -584,17 +592,30 @@ private boolean validateFilePath(File resourceFilePath) throws IOException {
   }
 
   protected WebClient getWebClient(String uri, Map<String, Serializable> properties) {
-    WebClient client;
+    SecureCxfClientFactory<WebClient> factory;
     if (properties.get(USERNAME) != null && properties.get(PASSWORD) != null) {
-      client =
-          WebClient.create(
-              uri, (String) properties.get(USERNAME), (String) properties.get(PASSWORD), null);
+      factory =
+          clientFactoryFactory.getSecureCxfClientFactory(
```

Comment:
Could also consider a single method that takes in a client configuration object.

Reply:
@peterhuffer I could be misinterpreting @coyotesqrl's comment, but that's what I was picturing. A clientConfiguration object with corresponding builder. In my opinion it would be _easier_ for callers, it would make the `ClientFactoryFactory` look a lot cleaner, and it would reduce the big nasty mocking blocks in the unit tests: https://github.com/codice/ddf/pull/3575/files/aa2174df245707f5f0ea2186c4897359b16c0db7#diff-1629f57032166e0d3e28cd49c0875c9bR412

Reply:
Hah. Yeah I knew I was going to get the builder comment. Honestly, I tried it and it seemed about the same to me as far as difficulty to read and it changed way more code. Also the issue that I had with it was that you can't mix and match all the different arguments, so you end up with a builder with a few big methods. I'm not going to change it in this PR because I'm trying to fix some stability issues right now, but it could get changed in the follow up one if enough people prefer the builder approach instead.